test/sql/index/art/constraints/test_art_concurrent_updates.test

```diff
@@ -0,0 +1,136 @@
+# name: test/sql/index/art/constraints/test_art_concurrent_updates.test
+# description: Test UPDATE in the main connection with concurrent updates in other connections.
+# group: [constraints]
+
+statement ok
+PRAGMA enable_verification
+
+statement ok
+CREATE TABLE tbl (id INT PRIMARY KEY, payload VARCHAR);
+
+statement ok
+INSERT INTO tbl VALUES (1, 'first payload'), (2, 'other payload');
+
+# con1 open.
+
+statement ok con1
+BEGIN TRANSACTION;
+
+statement ok con1
+UPDATE tbl SET payload = 'con1 payload' WHERE id = 1;
+
+statement ok con1
+UPDATE tbl SET id = 3 WHERE id = 2;
```

Comment:
Can we add an insert to `id=2` to verify that the update enables re-use of the primary key correctly?